### PR TITLE
Bulk updates

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Description of changes

Because the updates were easy, and I am not a fan of creating a bunch of branches just for each small change.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/jove/default.nix [hunk 1]
```diff
@@ -24,25 +25,34 @@ stdenv.mkDerivation (finalAttrs: {
     ncurses
   ];
 
+  postPatch = ''
+    patchShebangs testbuild.sh testmailer.sh teachjove jmake.sh
+    # substituteInPlace paths.h --replace "/bin/sh" "${runtimeShell}"
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jove/default.nix:30]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/txr/default.nix [hunk 2]
```diff
@@ -7,11 +7,11 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "txr";
-  version = "288";
+  version = "289";
 
   src = fetchurl {
     url = "http://www.kylheku.com/cgit/txr/snapshot/txr-${finalAttrs.version}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/tools/text/txr/default.nix:13]:
```suggestion
    url = "https://www.kylheku.com/cgit/txr/snapshot/txr-${finalAttrs.version}.tar.bz2";
```

File: pkgs/tools/misc/dialog/default.nix [hunk 3]
```diff
@@ -35,11 +40,11 @@ stdenv.mkDerivation (finalAttrs: {
...
   ];
 
-  meta = with lib; {
+  meta = {
     homepage = "https://invisible-island.net/dialog/dialog.html";
     description = "Display dialog boxes from shell";
-    license = licenses.lgpl21Plus;
-    maintainers = with maintainers; [ AndersonTorres spacefrogg ];
+    license = lib.licenses.lgpl21Plus;
+    maintainers = with lib.maintainers; [ AndersonTorres spacefrogg ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dialog/default.nix:47]: You don't need to change meta around, it just increases the diff

File: pkgs/tools/misc/dialog/default.nix [hunk 4]
```diff
@@ -3,25 +3,30 @@
...
+      "https://invisible-island.net/archives/dialog/dialog-${finalAttrs.version}.tgz"
+      "ftp://ftp.invisible-island.net/dialog/dialog-${finalAttrs.version}.tgz"
+    ];
+    hash = "sha256-DCYoIwUmS+IhfzNfN5j0ix3OPPEsWgdr8jHK33em1qg=";
   };
 
   buildInputs = [
     ncurses
+  ]
+  ++ lib.optionals withLibrary [
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dialog/default.nix:28]:
```suggestion
  ] ++ lib.optionals withLibrary [
```
nit

File: pkgs/tools/misc/dialog/default.nix [hunk 5]
```diff
@@ -3,25 +3,30 @@
...
   pname = "dialog";
-  version = "1.3-20220728";
+  version = "1.3-20230209";
 
   src = fetchurl {
-    url = "ftp://ftp.invisible-island.net/dialog/dialog-${finalAttrs.version}.tgz";
-    hash = "sha256-VEGJc9VZpGGwBpX6/mjfYvK8c9UGtDaCHXfKPfRUGQs=";
+    urls = [
+      "https://invisible-island.net/archives/dialog/dialog-${finalAttrs.version}.tgz"
+      "ftp://ftp.invisible-island.net/dialog/dialog-${finalAttrs.version}.tgz"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/dialog/default.nix:20]: We shouldn't use ftp as a fallback, it is very insecure.

File: pkgs/applications/office/vnote/default.nix [hunk 6]
```diff
@@ -1,37 +1,42 @@
...
-  version = "3.13.0";
+  version = "3.15.1";
 
   src = fetchFromGitHub {
     owner = "vnotex";
-    repo = pname;
+    repo = "vnote";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-A0OJzhvHP+muPI8N23zD4RTiyK0m3JGr/3uJ0Tqz97c=";
     fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/office/vnote/default.nix:20]:
```suggestion
    fetchSubmodules = true;
    hash = "sha256-A0OJzhvHP+muPI8N23zD4RTiyK0m3JGr/3uJ0Tqz97c=";
```
submodules change the hash, so it usually goes last

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please create one PR for each package.

File: pkgs/tools/misc/dialog/default.nix (hunk 5 above)
Comment (AndersonTorres) [reply pkgs/tools/misc/dialog/default.nix:20]: It was ftp originally. I will remove it.

File: pkgs/applications/office/vnote/default.nix (hunk 6 above)
Comment (AndersonTorres) [reply pkgs/applications/office/vnote/default.nix:20]: OK.

File: pkgs/tools/text/txr/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/tools/text/txr/default.nix:13]: Done

File: pkgs/tools/misc/dialog/default.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/tools/misc/dialog/default.nix:47]: I  decline, because of the harms of nested `with`s.

**Comment by AndersonTorres** [comment]
Closing because splits

https://github.com/NixOS/nixpkgs/pull/242584
https://github.com/NixOS/nixpkgs/pull/242583
https://github.com/NixOS/nixpkgs/pull/242582
https://github.com/NixOS/nixpkgs/pull/242581
https://github.com/NixOS/nixpkgs/pull/242580
https://github.com/NixOS/nixpkgs/pull/242579
https://github.com/NixOS/nixpkgs/pull/242578

**Pull request closed without merging**